### PR TITLE
Match "RECOMMENDS" to "SHOULD"

### PR DESCRIPTION
**Pull request by jmandel**
The specification states:
> This specification RECOMMENDS the use of `ECDH-ES` with the `X25519` curve for JWE
> as explained in section <a href="#encryption">Encryption</a> and described in

And then later in the encryption section states: 
>JWE encryption MUST use Diffie-Hellman key agreement, 

The "MUST" requirement is too strong, as it puts a requirement on the types of keys (e.g., EC keys vs RSA keys) an RP needs to support, as well as the specific curve for an EC key.

---
Also, even to make sense of the "RECOMMENDS" language: wouldn't  "`ECDH-ES` using the `X25519` curve" mean that the RP needs a `X25519` public key (vs the examples, which only show the RP providing a `secp256k1` public key). Is this point worth a follow-up PR to clarify?

### CODE REVIEW

**Comment by jmandel** [comment]
Thanks for the review. Re: my question here, should I open up another issue to discuss the inconsistency between examples and recommendations? Am I misinterpreting the implications of recommending a specific curve for the encryption protocol, with respect to key types allowed for the airline party?

**Comment by OR13** [comment]
https://safecurves.cr.yp.to/

https://crypto.stackexchange.com/questions/52983/why-is-there-the-option-to-use-nist-p-256-in-gnupg

TL;DR people don't trust the recommendations from IANA / Microsoft / NIST ... because of Snowden... trust is slow to build and quick to destroy... 

telling people to use MD5/Sha1 would be a mistake today... is telling people to pick from https://www.iana.org/assignments/jose/jose.xhtml

the same kind of mistake?

maybe... some people on the internet who some other people on the internet trust tend to recommend x25519 and ed25519 edsa these days... and to NOT recommend P-256....
